src/model.hpp

```diff
@@ -164,6 +164,9 @@ class Model {
          * @return specific model version
          */
     const std::shared_ptr<ModelInstance> getModelInstanceByVersion(const model_version_t& version) const {
+        if (version == 0) {
```

Comment:
Right now we have:
for TFS REST API:
we parse `request_path` & get requested model version there. If it is not requested we pass empty `std::optional` to the `processPredictRequest` -> `processSingleModelRequest` & we have `modelVersion.value_or(0)` when calling `modelManager.getModelInstance()`.

With this change you are effectively pushing the logic down into `Model` & we have two ways of handling that case -  1 for KFS & 1 for TFS. Either we cleanup the other ones or we should stick to calling `modelManager.getModelInstance` with `modelVersion.value_or(0)`.

Both solutions are not very clean so I would stick to the old one & refactor this in separate MR which would address this model version policy handling.